services/catalog/tests/unit/with_dbs/test_services_access_rights.py

```diff
@@ -0,0 +1,173 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+from typing import Callable, List
+
+import pytest
+from aiopg.sa.engine import Engine
+from fastapi import FastAPI
+from models_library.services import ServiceAccessRightsAtDB, ServiceDockerData
+from pytest_simcore.helpers.utils_mock import future_with_result
+from simcore_service_catalog.db.repositories.services import ServicesRepository
+from simcore_service_catalog.models.domain.group import GroupAtDB
+from simcore_service_catalog.services.access_rights import (
+    evaluate_auto_upgrade_policy,
+    evaluate_default_policy,
+    merge_access_rights,
+)
+
+pytest_simcore_core_services_selection = [
+    "postgres",
+]
+pytest_simcore_ops_services_selection = [
+    "adminer",
+]
+
+
+def test_merge_access_rights():
+    sample = ServiceAccessRightsAtDB.parse_obj(
+        {
+            "key": "simcore/services/dynamic/sim4life",
+            "version": "1.0.9",
+            "gid": 8,
+            "execute_access": True,
+            "write_access": True,
+            "product_name": "osparc",
+        }
+    )
+
+    # overrides and with other products
+    merged = merge_access_rights(
+        [
+            sample.copy(deep=True),
+            sample.copy(deep=True),
+            sample.copy(update={"execute_access": False}, deep=True),
+            sample.copy(update={"product_name": "s4l"}),
+        ]
+    )
+
+    assert len(merged) == 2
+    assert merged[0].get_flags() == {"execute_access": True, "write_access": True}
+
+    merged = merge_access_rights(
```

Comment:
What is this doing?

Reply:
check the implementation... :-)
there are several policies in place that might produce multiple access rights for the same resources. This merge function should resolve them. 
I am not very satisfied with the naming and th esolution ... I was expecting more feedback in the review :-(

Reply:
I refactored it a bit ... and now call it ``reduce_access_rights`` because is kind of a reduction operation on a list of access rights.